viabel/objectives.py

```diff
@@ -121,26 +130,148 @@ def __init__(self, approx, model, num_mc_samples, use_path_deriv=False):
             Number of Monte Carlo samples to use to approximate the objective.
         use_path_deriv : `bool`
             Use path derivative (for "sticking the landing") gradient estimator
+        hessian_approx_method : 'string'
+            Select from different methods for approximating the hessian:
+                'full' : use the full hessian matrix provided by BridgeStan
+                'mean_only' : use control variate only for mean estimator to avoid calculation of full hessian
+                'loo_diag_approx' : using "leave one out" method with hessian vector product value at other samples to
+                    estimate the diagonal values of hessian
+                'loo_direct_approx;: the same method as 'loo_diag_approx' but use the scaled approximation to the
+                    gradient of scale to do the "loo" estimation
         """
+
         self._use_path_deriv = use_path_deriv
+        if hessian_approx_method in [None, 'full', 'mean_only', 'loo_diag_approx', 'loo_direct_approx']:
+            self.hessian_approx_method = hessian_approx_method
+        else:
```

Comment:
Add one more test to validate that the constructor raises an error when in an invalid `hessian_approx_method` argument is passed in.